addons/survey/static/src/js/survey_form.js

```diff
@@ -142,20 +142,43 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
         }
     },
 
+    /**
+     * Handle visibility of comment area and conditional questions
+     * The form (page) is then automatically submitted if:
+     * - Survey is configured with one page per question and participants are allowed to go back,
+     * - It is not the last question of the survey,
+     * - The question is not waiting for a comment (with "Other" answer),
+     *
+     * @param event
+     */
+    _onChangeChoiceItem: function (event) {
+        const $target = $(event.currentTarget);
+        const $choiceItemGroup = $target.closest('.o_survey_form_choice');
+
+        this._renderCommentAreaVisibility($target);
+        const isQuestionComplete = this._applyAnswerSelectionChanges($target, $choiceItemGroup);
+        if (isQuestionComplete && this.options.usersCanGoBack) {
+            const isLastQuestion = this.$('button[value="finish"]').length !== 0;
+            if (!isLastQuestion) {
+                const questionHasComment = $target.hasClass('o_survey_js_form_other_comment') || $target
+                    .closest('.o_survey_form_choice')
+                    .find('.o_survey_comment').length !== 0;
+                if (!questionHasComment) {
+                    this._submitForm({});
+                }
+            }
+        }
+    },
     /**
     * Checks, if the 'other' choice is checked. Applies only if the comment count as answer.
     *   If not checked : Clear the comment textarea, hide and disable it
     *   If checked : enable the comment textarea, show and focus on it
     *
-    * @private
-    * @param {Event} event
+    * @param {JQuery<HTMLElement>} $choiceItemGroup
     */
-    _onChangeChoiceItem: function (event) {
-        var self = this;
-        var $target = $(event.currentTarget);
-        var $choiceItemGroup = $target.closest('.o_survey_form_choice');
-        var $otherItem = $choiceItemGroup.find('.o_survey_js_form_other_comment');
-        var $commentInput = $choiceItemGroup.find('textarea[type="text"]');
+    _renderCommentAreaVisibility: function ($choiceItemGroup) {
```

Comment:
It does not really "render" strictly speaking.

Maybe "_applyCommentAreaVisibility" ?